aiohttp/web_urldispatcher.py

```diff
@@ -1257,7 +1257,23 @@ def _quote_path(value: str) -> str:
 
 
 def _unquote_path(value: str) -> str:
-    return URL.build(path=value, encoded=True).path.replace("%2F", "/")
+    """Unquote a URL path.
+
+    This method is used to unquote %2F to / in the URL path
+    for paths returned from yarl.URL.path. It is not intended
+    for use with yarl.URL.raw_path.
+    """
+    # Note that older yarl versions already unquote "%2F" to "/"
+    # so we only want to unquote if we see "%2F" or "%2f"
```

Comment:
This still won't work right because the path segment can be quoted and we would be double decoding by doing this